src/main/daml/Daml/Finance/Claims/Lifecycle/Rule.daml

```diff
@@ -42,35 +43,25 @@ template Rule
     -- Lifecycle time update events
     interface instance Lifecycle.I for Rule where
       view = Lifecycle.View with lifecycler; id; description
-      -- BOND_PROCESS_CLOCK_UPDATE_INITAL_CLAIMS_BEGIN
+      -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_BEGIN
       evolve Lifecycle.Evolve{eventCid; observableCids; instrument} = do
-        claimInstrument <- fetchInterfaceByKey @BaseInstrument.R instrument
-      -- BOND_PROCESS_CLOCK_UPDATE_INITAL_CLAIMS_END
-        -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_FASTFORWARD_BEGIN
-        -- Recover claims tree as of the lastEventTimestamp. For a bond, this just requires
-        -- lifecycling as of the lastEventTimestamp.
-        nv <- BaseInstrument.exerciseInterfaceByKey @DynamicInstrument.I instrument lifecycler
-          DynamicInstrument.GetView with viewer = lifecycler
-        claims <- fst <$>
-          lifecycle lifecycler observableCids claimInstrument [timeEvent nv.lastEventTimestamp]
-        -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_FASTFORWARD_END
-        -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_BEGIN
-        -- Lifecycle
-        let acquisitionTime = Claim.getAcquisitionTime claimInstrument
         v <- view <$> fetch eventCid
-        (remaining, pending) <- lifecycleClaims observableCids acquisitionTime claims
-          [timeEvent v.eventTime]
+        (remaining, pending, claims, claimInstrument, nv) <-
```

Comment:
I would add a comment explaining that time-based lifecycling is happening at this stage